fix: prevent absolute file_path from overriding base save directory

johan-6 opened the pull request:
## 🐛 问题描述

`/upload-single/` 接口中，当 `file_path` 为绝对路径（例如以 `/` 开头）时，`os.path.join(wx_dir, file_path)` 会直接忽略前面的 `wx_dir`，导致文件被错误保存到服务器根目录（如 `/FileStorage/...`），而不是微信用户对应的 session 目录中。

## 🔁 复现步骤

调用上传接口 `/upload-single/`，传入如下 `file_path`：

```plaintext
/FileStorage/File/2025-04/xxx.pdf
```

查看日志，可见保存路径为：

```plaintext
保存路径：/FileStorage/File/2025-04/xxx.pdf
```

而不是期望中的：

```plaintext
保存路径：/app/data/sessions/{session_id}/wxid_{wx_id}/FileStorage/File/2025-04/xxx.pdf
```

## ✅ 解决方案

在拼接路径前，使用 `file_path.lstrip("/")` 将绝对路径转换为相对路径，确保其被正确拼接到 `wx_dir` 下：

```python
file_path = file_path.replace("\\", "/").lstrip("/")
save_path = os.path.join(wx_dir, file_path)
```

这样可以确保无论传入的路径是否带 `/`，文件都会保存在正确的微信用户目录中。

相关日志见附件图片
![image](https://github.com/user-attachments/assets/16bb6bab-eb89-4ff4-8864-f0533b9adfd1)

johan-6 committed:
fix: prevent absolute file_path from overriding base save directory

like-fly commented:
是的，这段代码确实有问题，有路径穿透的风险，不过即便这样改可能也不能完全规避，因为可以用`../abc.pdf`这样的方式，可能需要用 `.resolve()` 检查拼接后的目录是否仍然在 wx_dir 目录中。

johan-6 committed:
fix(upload): prevent directory traversal via file_path in /upload-single/

- Sanitized incoming file_path by stripping leading slashes and resolving the final save path.
- Added safety check to ensure that the resolved save path stays within the intended wx_dir.
- This prevents attackers from using paths like "../../etc/passwd" to write files outside the allowed directory.

johan-6 commented:
已增加路径穿透检测，再帮忙 review 下，谢谢！
